src/components/Filter/Filter.js

```diff
@@ -1,113 +1,116 @@
 import './Filter.css';
 
-import React, { Component } from 'react';
+import React, { useReducer } from 'react';
 import classNames from 'classnames';
 import AutoComplete from '../AutoComplete/AutoComplete';
 import Input from '../Input/Input';
 import Switch from '../Switch/Switch';
 
 import { generateLists } from '../../listsGenerator';
 
-export default class Filter extends Component {
-  state = {
-    tag: '',
-    country: '',
-    name: '',
-    language: '',
-    showFilters: false,
-  };
+const Filter = props => {
+  const [state, setState] = useReducer(
+    (state, newState) => ({ ...state, ...newState }),
```

Comment:
This is an interesting pattern.  What is the benefit of using a function as the state instead of sticking with the standard `dispatch` + `action.type` methodology?

Reply:
> What is the benefit of using a function as the state 

@brentmclark Not sure what you mean 😞 

The state is an object in this case. I went with this pattern because I wanted to keep it simple.

Reply:
I guess he meant for this: https://reactjs.org/docs/hooks-reference.html#usereducer

Reply:
@arku the typical `useReducer` setup -- as shown in the link provided by @moshfeu --  employs a `dispatch` + `action.type` methodology.  In your implementation, you're merging all of the new state onto the current state instead of using the approach outlined in the official documentation.  I am wondering if there is a benefit in deviating from the established pattern.  If there isn't a benefit to the users of the site, I think it would be wise to return to the established pattern so other devs can better understand what's happening in this code.

Reply:
@brentmclark As I mentioned in my another comment, simplicity was the only objective with this change. You're right that this deviates from the conventional pattern. How do you prefer we write this?

Reply:
Sorry for the delayed response, @arku; I was out of town.

Ideally this implementation would be reworked to bring it inline with the officially-documented approach.  More specifically, in a way that requires the use of action types wherever the reduced state is updated.